test/cri-containerd/container_test.go

```diff
@@ -136,3 +152,72 @@ func Test_RotateLogs_LCOW(t *testing.T) {
 		expected++
 	}
 }
+
+func Test_RunContainer_Events_LCOW(t *testing.T) {
+	pullRequiredLcowImages(t, []string{imageLcowK8sPause, imageLcowAlpine})
+	client := newTestRuntimeClient(t)
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	targetNamespace := "k8s.io"
+
+	sandboxRequest := &runtime.RunPodSandboxRequest{
+		Config: &runtime.PodSandboxConfig{
+			Metadata: &runtime.PodSandboxMetadata{
+				Name:      t.Name(),
+				Uid:       "0",
+				Namespace: testNamespace,
+			},
+		},
+		RuntimeHandler: lcowRuntimeHandler,
+	}
+
+	podID := runPodSandbox(t, client, ctx, sandboxRequest)
+	defer func() {
+		stopAndRemovePodSandbox(t, client, ctx, podID)
+	}()
+
+	request := &runtime.CreateContainerRequest{
+		Config: &runtime.ContainerConfig{
+			Metadata: &runtime.ContainerMetadata{
+				Name: t.Name() + "-Container",
+			},
+			Image: &runtime.ImageSpec{
+				Image: imageLcowAlpine,
+			},
+			Command: []string{
+				"top",
+			},
+			Linux: &runtime.LinuxContainerConfig{},
+		},
+		PodSandboxId:  podID,
+		SandboxConfig: sandboxRequest.Config,
+	}
+
+	topicNames, filters := getTargetRunTopics()
+	eventService := newTestEventService(t)
+	stream, errs := eventService.Subscribe(ctx, filters...)
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go runContainerWithWaitGroup(t, client, ctx, request, &wg)
+
+	for _, topic := range topicNames {
+		select {
+		case env := <-stream:
+			if topic != env.Topic {
+				t.Fatalf("event topic %v does not match expected topic %v", env.Topic, topic)
+			}
+			if targetNamespace != env.Namespace {
+				t.Fatalf("event namespace %v does not match expected namespace %v", env.Namespace, targetNamespace)
+			}
```

Comment:
In the case that we did get the expected event, I would do a `t.Log` to write out its topic and namespace. This won't cause the test to fail, and will be useful output (to see which events already occurred) when debugging test failures.

Reply:
Updated.